providers/src/airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -154,18 +156,20 @@ def execute(self, context: Context) -> GetStatementResultResponseTypeDef | str:
             session_keep_alive_seconds=self.session_keep_alive_seconds,
         )
 
-        self.statement_id = query_execution_output.statement_id
+        # Pull the statement ID, session ID
+        statement_id: str = query_execution_output.statement_id
+        self.statement_id = statement_id  # To be used in "on_kill" if needed
```

Comment:
No need to have 2 variables then, you could store it directly in `self.statement_id`